Backend/Underdogs-Struggling/src/main/java/com/TheTroisMousquetaires/Underdogs/Struggling/dao/EntryRepository.java

```diff
@@ -17,15 +21,20 @@ public interface EntryRepository extends CrudRepository<Entry, Long> {
 
     Optional<Entry> findEntryByTitle(String title);
 
-    List<Entry> findEntriesBySubtitle(String subtitle);
+    List<Entry> findEntriesByTopic(String topic);
 
     List<Entry> findEntriesByAuthor(User author);
 
-    List<Entry> findEntriesByTagsIn(Set<Tag> tags);
+    @Query("SELECT e FROM Entry e JOIN e.tags t WHERE t IN :tags GROUP BY e HAVING COUNT(DISTINCT t) = :tagCount")
+    List<Entry> findEntriesByTags(@Param("tags") Set<Tag> tags, @Param("tagCount") long tagCount);
```

Comment:
This one probably will not work. We do not store `tag` information in `Entries` table, entry-tag records are stored in the `EntryTags` table.